### PR TITLE
MonoState Pattern

### PR DESCRIPTION
**Pull request by amit2103**
Implemented the pattern. Created a custom load balancer. Will add java comments.
The Documentation and Class Diagram is present. Please let me know if any other things needs to be done.

### CODE REVIEW

**Comment by amit2103** [comment]
I will add the Junits, but I would like to now whether the design is okay

**Comment by iluwatar** [comment]
The initial code looks ok. The branch is not up to date with master. Please catch up the branch and implement the missing items in the following checklist.

- [X] Put under review badge to the pull request
- [X] Does the example code implement the pattern correctly and follow good coding practices?
- [ ] Does the example code have enough test coverage?
- [ ] Is the example code commented well enough?
- [ ] Is the example code following JavaDoc conventions?
- [ ] Are the project coding conventions being followed?
- [X] Is the class diagram generated correctly?
- [X] Is the index.md implemented correctly so the pattern will show correctly on the web site?

**Comment by npathai** [comment]
I am also reviewing it.

**Comment by amit2103** [comment]
@iluwatar  Thanks for the comment. Will get it done by this weekend.
@npathai  Let me know if you have any extra comments. i will incorporate them

**Comment by amit2103** [comment]
Don't know what caused the build break. I have updated the files with comments and Junits.
Please do a recheck when it is possible.

**Comment by iluwatar** [comment]
```[ERROR]     Non-resolvable parent POM: Failure to find com.iluwatar:java-design-patterns:pom:1.6.0 in http://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced and 'parent.relativePath' points at wrong local POM @ line 5, column 11 -> [Help 2]```

Use version number 1.7.0 in `pom.xml` to resolve this.

**Comment by amit2103** [comment]
@iluwatar  Thanks. Understood. I had already done a build with the parent POM at 1.6.0. So when I did  a local build yesterday, it passed since my local maven repo had a reference to 1.6.0. 

I have corrected the reference to the parent POM.

File: monostate/index.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+---
+
+**Intent:** Enforces a behaviour like sharing the same state amongst all instances.
+
+![alt text](./etc/monostate.png "MonoState")
+
+**Applicability:** Use the Monostate pattern when
+
+* The same state must be shared across all instances of a class.
+* Typically this pattern might be used everywhere a SingleTon might be used. Singleton usage however is not transparent, Monostate usage is.
```
Comment (iluwatar) [inline monostate/index.md]: should be "Singleton"

File: monostate/src/main/java/com/iluwatar/monostate/Request.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * 
+ * The Request class. A {@link Server} can handle an instance of a Request. 
+ * 
+ */
+
+public class Request {
+  public final String value;
+
+  public Request(String value) {
+    super();
```
Comment (iluwatar) [inline monostate/src/main/java/com/iluwatar/monostate/Request.java]: No need for `super()`

File: pom.xml [hunk 3]
```diff
@@ -73,8 +73,7 @@
 		<module>front-controller</module>
 		<module>repository</module>
 		<module>async-method-invocation</module>
-	    	<module>business-delegate</module>
-    		<module>half-sync-half-async</module>
```
Comment (iluwatar) [inline pom.xml]: These should not be removed, I think?

**Comment by iluwatar** [comment]
@npathai Any additional comments?

**Comment by amit2103** [comment]
@iluwatar I noticed the pom error was about to commit the changes with one minor modification. Committed the changes. Please do a review once you have time.

**Comment by npathai** [comment]
@iluwatar Had been away for some days. Will review it today.

**Comment by npathai** [comment]
Good work. Looks good to go. Thanks for your contribution.

**Comment by amit2103** [comment]
@npathai  Thanks...Will start working on the leader follower pattern

**Pull request merged**